### PR TITLE
Add kafka-payload-processor shaded module

### PR DESCRIPTION
**Pull request by BewareMyPower**
Fixes https://github.com/streamnative/kop/issues/958

### Motivation

Current `kafka-payload-processor` dependency is not compatible with `pulsar-client` dependency.

### Modifications

- Change the original `kafka-payload-processor` artifact id to `kafka-payload-processor-original`, which is compatible with `pulsar-client-original` dependency.
- Add a new module `kafka-payload-processor-shaded` whose artifact id is `kafka-payload-processor`. This module is compatible with `pulsar-client` dependency.
- Add a new module `kafka-payload-processor-shaded-tests` that contains a basic test for artifact `kafka-payload-processor` with `pulsar-client` dependency.

### CODE REVIEW

**Comment by BewareMyPower** [comment]
Now this PR is ready to review, PTAL @Demogorgon314

File: kafka-payload-processor-shaded-tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaPayloadProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.List;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.impl.MessageIdImpl;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class KafkaPayloadProcessorTest {
+
+    @Test
+    public void testValues() throws Exception {
```
Comment (Demogorgon314) [inline kafka-payload-processor-shaded-tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaPayloadProcessorTest.java:27]: Can we add some tests to verify bypassing the control batch?

Comment (BewareMyPower) [reply kafka-payload-processor-shaded-tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaPayloadProcessorTest.java:27]: This PR only tests the `kafka-payload-processor` dependency could work with `pulsar-client` dependency. So this test is more like a smoke test. I think we can use other PRs to add more tests, including tests for keys and headers.

**Review by Demogorgon314 — APPROVED** [review]
LGTM.

**Pull request merged**